cmd/image-factory/cmd/service.go

```diff
@@ -313,3 +339,48 @@ func loadPrivateKey(keyPath string) (crypto.PrivateKey, error) {
 
 	return cryptoutils.UnmarshalPEMToPrivateKey(fileBytes, cryptoutils.SkipPassword)
 }
+
+// Basically taken from https://github.com/sigstore/cosign/blob/main/cmd/cosign/cli/options/signature_digest.go
+func getHashAlgo(algo string) (crypto.Hash, error) {
+	var supportedSignatureAlgorithms = map[string]crypto.Hash{
+		"sha224": crypto.SHA224,
+		"sha256": crypto.SHA256,
+		"sha384": crypto.SHA384,
+		"sha512": crypto.SHA512,
+	}
+	normalizedAlgo := strings.ToLower(strings.TrimSpace(algo))
+
+	if normalizedAlgo == "" {
+		return crypto.SHA256, nil
+	}
+
+	ralgo, exists := supportedSignatureAlgorithms[normalizedAlgo]
+	if !exists {
+		return crypto.SHA256, fmt.Errorf("unknown digest algorithm: %s", algo)
+	}
+
+	if !ralgo.Available() {
+		return crypto.SHA256, fmt.Errorf("hash %q is not available on this platform", algo)
+	}
+	return ralgo, nil
+}
+
+func getPublicKeyVerifier(opts Options) (sigstoresignature.Verifier, error) {
+	var keyVerifier sigstoresignature.Verifier
```

Comment:
this code and many others won't pass `make lint`

Reply:
Fixed up the code so that make lint passes.